scripts/plot_balance_map_interactive.py

```diff
@@ -0,0 +1,258 @@
+"""
+
+import geopandas as gpd
+import matplotlib.colors as mcolors
+import matplotlib.pyplot as plt
+import pydeck as pdk
+import pypsa
+from pypsa.plot.maps.interactive import PydeckPlotter
+from pypsa.statistics import get_transmission_carriers
+
+from scripts._helpers import (
+    configure_logging,
+    set_scenario_config,
+    update_config_from_wildcards,
+)
+from scripts.add_electricity import sanitize_carriers
+
+VALID_MAP_STYLES = PydeckPlotter.VALID_MAP_STYLES
+
+
+def scalar_to_rgba(
+    value: float,
+    *,
+    norm: mcolors.Normalize,
+    cmap: mcolors.Colormap,
+    alpha: float = 1.0,
+) -> list[int]:
+    """
+    Map a scalar float value to an RGBA color encoded as 8-bit integers.
+
+    Parameters
+    ----------
+    value : float
+        Scalar input to map through the normalization and colormap.
+    norm : matplotlib.colors.Normalize
+        Normalization defining vmin and vmax used for scaling.
+    cmap : matplotlib.colors.Colormap
+        Colormap used to convert normalized values to RGBA colors.
+    alpha : float, optional (default = 1.0)
+        Opacity in the range [0, 1]. Overrides the colormap's alpha.
+
+    Returns
+    -------
+    List[int]
+        A list ``[R, G, B, A]`` where each channel is an integer in the 0–255 range.
+    """
+
+    # Clamp to normalization bounds
+    p = max(norm.vmin, min(norm.vmax, value))
+
+    # Convert to RGBA floats (0–1)
+    r, g, b, _ = cmap(norm(p))
+
+    # Clamp and apply alpha
+    a = alpha if 0.0 <= alpha <= 1.0 else 1.0
+
+    # Convert to 8-bit integers
+    return [
+        int(r * 255),
+        int(g * 255),
+        int(b * 255),
+        int(a * 255),
+    ]
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        from scripts._helpers import mock_snakemake
+
+        snakemake = mock_snakemake(
+            "plot_balance_map_interactive",
+            clusters=50,
+            opts="",
+            sector_opts="",
+            planning_horizons="2050",
+            carrier="H2",
+        )
+
+    configure_logging(snakemake)
+    set_scenario_config(snakemake)
+    update_config_from_wildcards(snakemake.config, snakemake.wildcards)
+
+    # Interactive map settings
+    settings = snakemake.params.settings
+    conversion = settings.get("conversion") or 1
+    region_cmap = settings.get("region_cmap") or "Reds"
+    region_alpha = settings.get("region_alpha") or 0.7
+    region_unit = settings.get("region_unit") or ""
+    branch_color = settings.get("branch_color") or "darkseagreen"
+    arrow_size_factor = settings.get("arrow_size_factor") or 1.5  # PyPSA default
+    bus_size_max = settings.get("bus_size_max") or 10000  # PyPSA default
+    branch_width_max = settings.get("branch_width_max") or 10  # PyPSA default
+    map_style = (str(settings.get("map_style")) or "road").lower()
+    map_style = VALID_MAP_STYLES.get(map_style, "road")
+    tooltip = settings.get("tooltip") or True
```

Comment:
With our setting you can write code assuming that the config settings exist. Defaults are all handled in `config.default.yaml`.